pulsar-broker-common/src/main/java/org/apache/pulsar/broker/resources/ClusterResources.java

```diff
@@ -101,11 +102,21 @@ public CompletableFuture<Void> deleteClusterAsync(String clusterName) {
         return deleteAsync(joinPath(BASE_CLUSTERS_PATH, clusterName));
     }
 
-    public CompletableFuture<Boolean> isClusterUsedAsync(String clusterName) {
+    public CompletableFuture<Boolean> isClusterUsedAsync(String currentClusterName, String clusterName) {
         return getCache().getChildren(BASE_POLICIES_PATH)
                 .thenCompose(tenants -> {
-                    List<CompletableFuture<List<String>>> futures = tenants.stream()
-                            .map(tenant -> getCache().getChildren(joinPath(BASE_POLICIES_PATH, tenant, clusterName)))
+                    List<CompletableFuture<List<Object>>> futures = tenants.stream()
+                            .map(tenant -> getCache().getChildren(joinPath(BASE_POLICIES_PATH, tenant, clusterName))
+                                .thenCombineAsync(
+                                        currentClusterName.equalsIgnoreCase(clusterName)
+                                                ? getCache().getChildren(joinPath(BASE_POLICIES_PATH, tenant)) :
+                                        CompletableFuture.completedFuture(new ArrayList()),
+                                    (v1, v2) -> {
+                                        List<Object> ret = new ArrayList<>(v1);
+                                        ret.addAll(v2);
+                                        return ret;
+                                    })
+                            )
```

Comment:
```suggestion
                            .map(tenant -> {
                                if (currentClusterName.equalsIgnoreCase(clusterName)) {
                                   return getCache().getChildren(joinPath(BASE_POLICIES_PATH, tenant));
                                }
                                return getCache().getChildren(joinPath(BASE_POLICIES_PATH, tenant, clusterName));
                            })
```

This will be more clear and efficient